### ISSUE TITLE
Memory access fault in miopenGemm

### ISSUE THREAD

**Issue by patflick** [ROCmSoftwarePlatform/MIOpen#8]
There's a bug inside the miopenGemm function call resulting in a memory access fault. 
I've tried debugging it a bit. So far I found that the kernel accesses the A matrix one element past its size, ie, it accesses `A[a_data_size]`.

The issue can be reproduced with the `MIOpenDriver`, which rules out the cause in my code:
```
$ ./MIOpenDriver gemm -m 128 -k 9216 -n 4096 -v 0
MIOpenDriver: gemm -m 128 -k 9216 -n 4096 -v 0
Memory access fault by GPU node-1 on address 0x1101808000. Reason: Page not present or supervisor privilege.
Aborted (core dumped)
```

This may be an issue inside [MIOpenGemm](https://github.com/ROCmSoftwarePlatform/MIOpenGEMM).

**Comment by dagamayank** [comment]
/cc @newling

@patflick I can reproduce this bug. Let us try to debug. Thanks for reporting.

**Comment by newling** [comment]
@dagamayank 
It's failing through the driver for certain not (m = n = k) cases. I'm investigating.

**Comment by newling** [comment]
The problem is related to not switching pointers  a <-> b when you switch rowMajor <-> colMajor.  Pointers must switch (along with lda <-> ldb etc).

**Comment by newling** [comment]
@dagamayank 
A solution might be to swap `DataCast(A)` and `DataCast(B)` on line 62 of gemm_api.cpp when rowMaj but this is probably not the neatest solution as MIOpenGEMM already deals with rowMaj and colMaj. I'll need to think more.

**Comment by dagamayank** [comment]
Thanks @newling and @patflick We have a fix for this bug and will push it out with the next release of MIOpen.

**Comment by patflick** [comment]
Is it possible to get the fix beforehand? I'm already compiling MiOpen myself. Ie, could you push the fix to some branch?

**Comment by dagamayank** [comment]
Fixed by 78955d5360ceb221516abe9490d48f3b726929a8 and 64e7553fe72ed8705119f8276b39b38886ffb3a7
